test/component/new_post_test.dart

```diff
@@ -19,45 +18,51 @@ import "../services/mock_geo_location_service.dart";
 import "../services/test_data/firestore_user_mock.dart";
 
 void main() {
-  setupFirebaseAuthMocks();
-
-  MockPostRepositoryService postRepository = MockPostRepositoryService();
-  MockGeoLocationService geoLocationService = MockGeoLocationService();
+  late MockPostRepositoryService postRepository;
+  late MockGeoLocationService geoLocationService;
+  late ProviderScope mockedNewPostPage;
 
   const timeDeltaMils = 500;
 
-  setUpAll(() async {
-    await Firebase.initializeApp();
+  setUp(() async {
+    setupFirebaseAuthMocks();
+
+    postRepository = MockPostRepositoryService();
+    geoLocationService = MockGeoLocationService();
+
+    mockedNewPostPage = ProviderScope(
+      overrides: [
+        ...firebaseMocksOverrides,
+        postRepositoryProvider.overrideWithValue(postRepository),
+        geoLocationServiceProvider.overrideWithValue(geoLocationService),
```

Comment:
I found that `MockGeoLocaitonService` is defined in `geolocation_service_test.dart`. I think it would make sense to define in its own file, together with its override, and to use this here instead.

Reply:
This will go in another sprint task dedicated to overrides and mocked pages :)